bin/idyll.js

```diff
@@ -1,53 +1,67 @@
+  .usage('Usage: idyll index.idl')
+  .example('$0 index.idl --build', 'Turn index.idl into output')
+  .demandCommand(1)
+  .alias({
+    b: 'build',
+    m: 'components',
+    c: 'css',
+    d: 'datasets',
+    g: 'debug',
+    q: 'defaultComponents',
+    f: 'inputFile',
+    s: 'inputString',
+    l: 'layout',
+    o: 'output',
+    k: 'spellcheck',
+    t: 'template',
+    e: 'theme'
+  })
+  .boolean('build')
+  .describe('build', 'Generate output files and exit')
+  .default('build', false)
+  .describe('components', 'Directory where components are located')
+  .default('components', 'components')
+  .describe('css', 'Custom CSS file to include in output')
+  .describe('datasets', 'Directory where data files are located')
+  .default('datasets', 'data')
+  .boolean('debug')
+  .describe('debug', 'Skip JS minification')
+  .default('debug', false)
+  .describe('defaultComponents', 'Directory where default set of components are located')
+  .default('defaultComponents', 'components/default')
+  .describe('inputFile', 'File containing Idyll source')
+  .describe('inputString', 'Idyll source as a string')
+  .describe('layout', 'Name of (or path to) the layout to use')
+  .default('layout', 'blog')
+  .describe('output', 'Directory where built files should be written')
+  .default('output', 'build')
+  .boolean('spellcheck')
+  .default('spellcheck', true)
+  .describe('spellcheck', 'Check spelling of Idyll source input')
+  .describe('template', 'Path to HTML template')
+  .default('template', '_index.html')
+  .describe('theme', 'Name of (or path to) the theme to use')
+  .default('theme', 'idyll')
+  .alias('h', 'help')
+  .argv;
+
+// inputFile can be passed as non-hypenated argument
+if (argv._[0]) argv.f = argv.inputFile = argv._[0];
 
-let options = {
-  compilerOptions: {
-    spellcheck: argv.spellcheck
-  },
-  build: argv.build ? argv.build : false
+// move spellcheck down a level
+argv.compilerOptions = {
+  spellcheck: argv.spellcheck
 };
 
-if (argv.hasOwnProperty('output')) {
-  options.output = argv.output;
-}
-if (argv.hasOwnProperty('template')) {
-  options.htmlTemplate = argv.template;
-}
-if (argv.hasOwnProperty('css')) {
-  options.css = argv.css;
-}
-if (argv.hasOwnProperty('components')) {
-  options.componentFolder = argv.components;
-}
-if (argv.hasOwnProperty('default-components')) {
-  options.defaultComponents = argv['default-components'];
-}
-if (argv.hasOwnProperty('datasets')) {
-  options.dataFolder = argv.datasets;
-}
-if (argv.hasOwnProperty('layout')) {
-  options.layout = argv.layout;
-  if (isPath(options.layout)) {
-    options.layout = path.resolve(options.layout);
-  }
-}
-if (argv.hasOwnProperty('theme')) {
-  options.theme = argv.theme;
-  if (isPath(options.theme)) {
-    options.theme = path.resolve(options.theme);
-  }
-}
+// delete stuff we don't need
+delete argv._;
+delete argv['$0'];
+delete argv.k;
+delete argv.spellcheck;
 
-idyll(path.resolve(argv._[0]), options);
```

Comment:
I added this `path.resolve` in to resolve some weird bug when using idyll linked locally. Will test this to make sure everything still works.